BGL/include/CGAL/boost/graph/io.h

```diff
@@ -0,0 +1,202 @@
+// Copyright (c) 2015  GeometryFactory (France).  All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org); you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public License as
+// published by the Free Software Foundation; either version 3 of the License,
+// or (at your option) any later version.
+//
+// Licensees holding a valid commercial license may use this file in
+// accordance with the commercial license agreement provided with the software.
+//
+// This file is provided AS IS with NO WARRANTY OF ANY KIND, INCLUDING THE
+// WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
+//
+// $URL$
+// $Id$
+//
+//
+// Author(s)     : Andreas Fabri
+
+#ifndef CGAL_BOOST_GRAPH_IO_H
+#define CGAL_BOOST_GRAPH_IO_H
+
+#include <CGAL/boost/graph/Euler_operations.h>
+#include <CGAL/boost/graph/helpers.h>
+#include <iostream>
+#include <algorithm>
+#include <sstream>
+#include <fstream>
+#include <string>
+#include <boost/foreach.hpp>
+#include <boost/container/flat_map.hpp>
+
+namespace CGAL {
+/*!
+   \ingroup PkgBGLIOFct
+    writes the graph `g` in the OFF format.
+    \sa Overloads of this function for specific models of the concept `FaceGraph`.
+
+  */ 
+template <typename FaceGraph>
+bool write_off(std::ostream& os,
+               const FaceGraph& g)
+{
+  typedef typename boost::graph_traits<FaceGraph>::vertex_descriptor vertex_descriptor;
+  typedef typename boost::graph_traits<FaceGraph>::face_descriptor face_descriptor;
+  typedef typename boost::graph_traits<FaceGraph>::vertices_size_type vertices_size_type;
+  typedef typename boost::graph_traits<FaceGraph>::faces_size_type faces_size_type;
+
+  typename boost::property_map<FaceGraph, CGAL::vertex_point_t>::const_type vpm = get(CGAL::vertex_point,g);
+  vertices_size_type nv = static_cast<vertices_size_type>(std::distance(vertices(g).first, vertices(g).second));
+  faces_size_type nf = static_cast<faces_size_type>(std::distance(faces(g).first, faces(g).second));
+
+  os << "OFF\n" << nv << " " << nf << " 0\n";
+  boost::container::flat_map<vertex_descriptor,vertices_size_type> reindex;
+  int n = 0;
+  BOOST_FOREACH(vertex_descriptor v, vertices(g)){
+    os << get(vpm,v) << '\n';
+    reindex[v]=n++;
+  }
+  
+  BOOST_FOREACH(face_descriptor f, faces(g)){
+    os << degree(f,g);
+    BOOST_FOREACH(vertex_descriptor v, vertices_around_face(halfedge(f,g),g)){
+      os << " " << reindex[v];
+    }
+    os << '\n';
+  }
+  return os.good();
+}
+
+
+/*!
+   \ingroup PkgBGLIOFct
+    writes the graph `g` in the OFF format.
+    \sa Overloads of this function for specific models of the concept `FaceGraph`.
+
+  */ 
+template <typename FaceGraph>
+bool write_off(const char* fname,
+               const FaceGraph& g)
+{
+  std::ofstream out(fname);
+  if(out.good()){
+    return write_off(out,g);
+  }
+  return false;
+}
+
+
+  namespace internal { namespace read_off {
```

Comment:
This namespace should be renamed (for example into read_off_tools ?) to avoid a conflict with the function read_off (an error was raised by CentOS6 in the test suite)